packages/api/src/payments/stripePaymentProvider.ts

```diff
@@ -83,36 +86,53 @@ export class StripePaymentProvider extends BasePaymentProvider {
     return intentStates
   }
 
-  async createIntent(props: CreatePaymentIntentProps): Promise<Intent> {
-    let paymentMethod: Stripe.PaymentMethod | undefined
-    if (props.customerID) {
-      const paymentMethods = await this.stripe.paymentMethods.list({
-        customer: props.customerID,
-        type: 'card'
-      })
-      paymentMethod = paymentMethods.data.length > 0 ? paymentMethods.data[0] : undefined
+  async createIntent({
+    customerID,
+    invoice,
+    saveCustomer,
+    paymentID
+  }: CreatePaymentIntentProps): Promise<Intent> {
+    let paymentMethod: Stripe.PaymentMethod | string | undefined
+    if (customerID) {
+      // For an off_session payment the default_payment_method of the customer will be used.
+      // If no user, deleted user or no default_payment_method the intent will be created without an customer.
+      const customer = await this.stripe.customers.retrieve(customerID)
+      if (customer.deleted) {
+        logger('stripePaymentProvider').warn(
+          'Provided customerID "%s" returns a deleted stripe customer',
+          customerID
+        )
+      }
+      if (!customer.deleted && customer.invoice_settings.default_payment_method !== null) {
+        paymentMethod = customer.invoice_settings.default_payment_method
+      } else {
+        logger('stripePaymentProvider').warn(
+          'Provided customerID "%s" has no default_payment_method',
+          customerID
+        )
+      }
     }
 
     const intent = await this.stripe.paymentIntents.create({
-      amount: props.invoice.items.reduce(
+      amount: invoice.items.reduce(
         (prevItem, currentItem) => prevItem + currentItem.amount * currentItem.quantity,
         0
       ),
-      ...(props.customerID
+      ...(customerID && paymentMethod // only add customer if we have customer and paymentMethod
         ? {
             confirm: true,
-            customer: props.customerID,
+            customer: customerID,
             off_session: true,
-            payment_method: paymentMethod?.id,
+            payment_method: typeof paymentMethod === 'string' ? paymentMethod : paymentMethod?.id,
```

Comment:
Took me a while you expect this multi-type paymentMethod - that's historical then. If this is always an `string` `id`, I would keep it as `string`/`null` only.

Reply:
Plus even rename this to `paymentMethodId`.